include/rtweekend.hpp

```diff
@@ -91,6 +91,18 @@ class LocalPseudoRNG {
   xorshift<> generator;
 };
 
+template<typename T>
+struct task_context {
+  LocalPseudoRNG rng;
+  T texture_data;
+};
+
+template<typename T>
+task_context<T> make_context(LocalPseudoRNG& rng, T& texture_data) {
```

Comment:
I would rather use a constructor with CTAD

Reply:
Indeed that is more modern.

Reply:
Not needed anymore due to the use of `sycl::global_ptr` instead of accessore